### ISSUE TITLE
Include dir

### ISSUE THREAD

**Issue by spiffyjr** [air-verse/air#40]
Would it be possible to add include_dir which would be used instead of exclude_dir. We have a monolithic repository and it would be easiest to have it watch the root and only include directories the project needs.  Thanks!

**Comment by cosmtrek** [comment]
OK, I’ll implement it:)

**Comment by spiffyjr** [comment]
Great, thanks!

**Comment by cosmtrek** [comment]
@spiffyjr Done, could you try the latest version?

**Comment by spiffyjr** [comment]
I tried it out but adding `root = "../"` seems to not watch anything and I was able to get `include_dir` to work with it.

**Comment by cosmtrek** [comment]
Suppose the repo's directory:
```
repo
|-- A
      |-- A1
      |-- A2
|-- B
      |-- B1
      |-- B2
```

If you want to watch `A`, the air config should like this:
```
root = "."
include_dir = ["A"]
...
```
and running `air` in the root directory of repo should work.

**Comment by spiffyjr** [comment]
Yeah, but if I want the root to be `A1` (where I run air) and to watch `A` it doesn't seem to work?

`root = "../../"
include_dir = ["A"]
...
```

**Comment by cosmtrek** [comment]
@spiffyjr seems wired and not work...

**Comment by spiffyjr** [comment]
I'm not sure what you mean @cosmtrek ?

**Comment by spiffyjr** [comment]
See https://github.com/cosmtrek/air/pull/43. This does what I need but I'm not sure it conflicts with anything else.

**Comment by spiffyjr** [comment]
#43 didn't quite do what I needed in all circumstances but #44 should do the trick.

**Comment by SimonPrague** [comment]
So is this working? I tried to watch submodule but it did not work.

**Comment by spiffyjr** [comment]
This was working when I submitted the pull request, but no longer merges with the latest master. I'd still like to see this added because the latest master no longer works with project structure.

**Comment by finnng** [comment]
I got into this similar issue
```
repo structure
|-- A
      |-- A1
      |-- A2
|-- B (air run in this folder)
      |-- .air.toml
      |-- main.go
```
I config include_dir like this but it didn't watch the `../A`
```
  include_dir = [root, "../A"]
```

**Comment by xiantang** [comment]
close for stale issue
